### ISSUE TITLE
Occasionally, no symbolicated backtrace available & '(unknown)' in the item description

### ISSUE THREAD

**Issue by and-megan** [rollbar/rollbar-ios#52]
As we have been testing it, and deliberately submitting crashes, we observe that sometimes there is no symbolicated backtrace available. It also seems that those reports contain '(unknown)' in the item description.

The detailed information on the occurrences indicates "This backtrace could not be symbolicated because a matching dSYM upload was not present" but there is a dSYM with the correct version uploaded. 

https://app.intercom.io/a/apps/rlisg82q/respond/inbox/813334/conversations/12600046934

**Comment by rokob** [comment]
We need to have more solid documentation about how symbolication works, when it doesn't work, and ensure that we are getting the right dSYMs. A lot of this issue is probably related to our backend symbolication infrastructure. However, if we move to KSCrash we should attempt to fix this on the SDK side by using on-device symbolication as much as possible. Therefore, adding this to the v1 milestone to track that use case.

**Comment by sterlingwes** [comment]
I'm running into this right now... seems the docs still need some love?

**Comment by jessewgibbs** [comment]
@sterlingwes

**Comment by akornich** [comment]
@jessewgibbs , so what exactly do we want to do about this one? Unfortunately, I have no access to the Intercom link provided above.

**Comment by sterlingwes** [comment]
example of what she's referring to, I think:

<img width="1242" alt="screen shot 2019-01-11 at 1 53 42 pm" src="https://user-images.githubusercontent.com/1047502/51053890-ff189180-15a8-11e9-8717-bc0d60e5e7f5.png">

I'll be opening another issue soon with a reproduction repo hopefully to demonstrate the other issues I'm having getting iOS crashes to symbolicate properly. There's either a step we're all missing b/c the docs are inadequate or the crash reports don't surface the right information. From my end though, it's been a struggle to get the kind of crash reports in rollbar that i'm accustomed to in crashlytics (where at the very least, my project's frames don't get collapsed as "non-project frames" and there's actually method / line number references instead of just addresses, per below where my project is `Wealthsimple`)

<img width="678" alt="screen shot 2019-01-11 at 2 01 58 pm" src="https://user-images.githubusercontent.com/1047502/51054060-87973200-15a9-11e9-9893-fbac87ee9267.png">

both crashes in these screens show as having matched against my dSym upload under "occurrences"

https://rollbar.com/wealthsimple/mobile/items/19/?item_page=0&item_count=100&#traceback
https://rollbar.com/wealthsimple/mobile/items/23/?item_page=0&item_count=100&#traceback

**Comment by akornich** [comment]
@sterlingwes , looks like #148 (that is closely related to this one) was closed. Can we close this one as well or do you need more follow-up on this one?

**Comment by sterlingwes** [comment]
@akornich we're still unfortunately getting unhelpful traces:
for example: https://rollbar.com/wealthsimple/mobile/items/657

You'll see in the traceback that it collapses our project frame `Wealthsimple`

**Comment by akornich** [comment]
@sterlingwes , thanks for the prompt reply!
let's ask @rokob about an update on the status of this one. last time we discussed it, this issue was classified as a backend issue. so i am not sure what was the priority/plan for addressing it there.

**Comment by danielsimons1** [comment]
I am still having this issue. Here is an example: https://rollbar.com/daniel.simons1/Score-Darts/items/2/occurrences/67861353793/

**Comment by jessicahsieh** [comment]
@rokob issue has come up again with another customer

internal link - https://app.intercom.io/a/apps/rlisg82q/inbox/inbox/488687/conversations/21255506888

**Comment by mattmikolay** [comment]
I am experiencing this issue as well for native code crashes on iOS recorded by the Rollbar React Native SDK. Is there any ETA for a fix? Thanks.

**Comment by jessicahsieh** [comment]
issue has come up for iPhone XR as well

internal link - https://app.intercom.io/a/apps/rlisg82q/inbox/inbox/488687/conversations/21981316924

**Comment by akornich** [comment]
#160

**Comment by letaniaferreira** [comment]
Maybe related - https://app.intercom.io/a/apps/rlisg82q/inbox/inbox/488687/conversations/22159921478

**Comment by letaniaferreira** [comment]
another customer affected - https://app.intercom.io/a/apps/rlisg82q/inbox/inbox/813342/conversations/24048857931

**Comment by kevinmcconnell** [comment]
Is there any news about progress on this issue?  It's happening to my company's applications now, and without symbolication the Rollbar reports are largely useless.

We've moved our affected iOS applications to Firebase Crashlytics (which is handling symbolication flawlessly so far) but it would be great to get an update on any possible fixes before we move everything else.

**Comment by akornich** [comment]
Hi @kevinmcconnell, may I ask you to provide a link to an occurrence that has the problem? Or you can post it for our support team on Intercom...

**Comment by kevinmcconnell** [comment]
@akornich sure!  Here's one where the stack trace says symbolicated, but has no symbols for the project code: https://rollbar.com/Arcivr/ceremony-booth/items/6/occurrences/109239617038/

And here's one that also has the `(unknown)` title: https://rollbar.com/Arcivr/ceremony-booth/items/3/occurrences/110265318396/

Thanks.

**Comment by akornich** [comment]
Hi @kevinmcconnell , thanks for posting the samples. Your SDK configuration looks fine and the expected dSYMs are uploaded to the project.  I contacted our payload processing team to look into why the dSYMs are not applied in these cases. Meanwhile, did you ever try [on-device symbolication](https://docs.rollbar.com/docs/ios#section-enabling-on-device-symbolication-from-ks-crash)?

**Comment by prat14k** [comment]
Hi @akornich, I am also facing similar issue. I have generated a crash 3-4 times, but only once is it is shown as symbolicated, rest all occurrences comes under "unknown". Note: 
    - The dsym upload is successful. The dsym for the build number can be seen on the rollbar panel
    - Any other DSYM for the same build number gives request timeout
    - The crash is generated in the same build install as of the dsym upload
    - Build Config: Release
    - Dwarf-with-dsym file is selected in Debug Information format in Build Settings
    - Debugging Symbols is selected in Strip Style in Build Settings (for on-device symbolication)
    - Server Post token used in the Python Run Script

I had to update the dsym upload python script as PlistBuddy command was not able to get the correct build number in Xcode 11. This is the updated script: 
`
import os
import subprocess
import zipfile

if (os.environ['DEBUG_INFORMATION_FORMAT'] != 'dwarf-with-dsym' or
        os.environ['EFFECTIVE_PLATFORM_NAME'] == '-iphonesimulator'):
    exit(0)

ACCESS_TOKEN = '508d93b4a658420bb533d80ae38c1bbb'

dsym_file_path = os.path.join(os.environ['DWARF_DSYM_FOLDER_PATH'],
                              os.environ['DWARF_DSYM_FILE_NAME'])
zip_location = '%s.zip' % (dsym_file_path)

os.chdir(os.environ['DWARF_DSYM_FOLDER_PATH'])
with zipfile.ZipFile(zip_location, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk(os.environ['DWARF_DSYM_FILE_NAME']):
        zipf.write(root)

        for f in files:
            zipf.write(os.path.join(root, f))

# You may need to change the following path to match your application
# settings and Xcode version
version = os.environ['CURRENT_PROJECT_VERSION']

curl_command = ('curl -X POST "https://api.rollbar.com/api/1/dsym" '
                '-F access_token=%s -F version=%s -F bundle_identifier="%s" '
                '-F dsym=@"%s"')
p = subprocess.Popen(curl_command % (ACCESS_TOKEN, version,
                     os.environ['PRODUCT_BUNDLE_IDENTIFIER'], zip_location),
                     shell=True)
p.communicate()
`

**Comment by prat14k** [comment]
Hi @akornich, I am also facing similar issue. I have generated a crash 3-4 times, but only once is it is shown as symbolicated, rest all occurrences comes under "unknown". Note:
- The dsym upload is successful. The dsym for the build number can be seen on the rollbar panel
- Any other DSYM for the same build number gives request timeout
- The crash is generated in the same build install as of the dsym upload
- Build Config: Release
- Dwarf-with-dsym file is selected in Debug Information format in Build Settings
- Debugging Symbols is selected in Strip Style in Build Settings (for on-device symbolication)
- Server Post token used in the Python Run Script

I had to update the dsym upload python script as PlistBuddy command was not able to get the correct build number in Xcode 11. This is the updated script:
```
import os
import subprocess
import zipfile

if (os.environ['DEBUG_INFORMATION_FORMAT'] != 'dwarf-with-dsym' or
        os.environ['EFFECTIVE_PLATFORM_NAME'] == '-iphonesimulator'):
    exit(0)

ACCESS_TOKEN = 'SERVER_TOKEN'

dsym_file_path = os.path.join(os.environ['DWARF_DSYM_FOLDER_PATH'],
                              os.environ['DWARF_DSYM_FILE_NAME'])
zip_location = '%s.zip' % (dsym_file_path)

os.chdir(os.environ['DWARF_DSYM_FOLDER_PATH'])
with zipfile.ZipFile(zip_location, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk(os.environ['DWARF_DSYM_FILE_NAME']):
        zipf.write(root)

        for f in files:
            zipf.write(os.path.join(root, f))

version = os.environ['CURRENT_PROJECT_VERSION']

curl_command = ('curl -X POST "https://api.rollbar.com/api/1/dsym" '
                '-F access_token=%s -F version=%s -F bundle_identifier="%s" '
                '-F dsym=@"%s"')
p = subprocess.Popen(curl_command % (ACCESS_TOKEN, version,
                     os.environ['PRODUCT_BUNDLE_IDENTIFIER'], zip_location),
                     shell=True)
p.communicate()
```

**Comment by akornich** [comment]
Hi @prat14k, thanks for reporting the issue. Let me rephrase your problem - just to make sure there is no misunderstanding.
You are using on-device symbolication and still getting it applied only occasionally? Are all the instances, that appear not symbolicated, the crash reports (or they are trace payloads)?
Also, may I ask you to open a support ticket on [Intercom](https://app.intercom.io/) and to provide there the links to the similar occurrences that do have and do not have symbolication working?

**Comment by akornich** [comment]
@prat14k , btw, what version of the SDK are you currently using?

**Comment by prat14k** [comment]
@akornich 

- Yes, I am using on-device symbolication and still getting it applied only occasionally
- All other instances, matching DSYMs are not found (as mentioned in the occurrence detail page). They should be using the last pushed DSYMs as the same installed build is being used for generating the crash. 
- I am using version 1.11.1.

Yes, I will open a support ticket on Intercom and mention the details there.

**Comment by prat14k** [comment]
@akornich  I am having confusion as to how open a support ticket on Intercom for Rollbar. So, I'll just post the links here.

- Occurence for which matching DSYM's are found:
         - https://rollbar.com/prateek-5/member-ios-app/items/5/occurrences/111603443492/
- Unmatched DSYM occurence of the same issue:
         - https://rollbar.com/prateek-5/member-ios-app/items/3/occurrences/111603505634/

Let me know if you need anything else from me.

**Comment by kevinmcconnell** [comment]
> Meanwhile, did you ever try [on-device symbolication](https://docs.rollbar.com/docs/ios#section-enabling-on-device-symbolication-from-ks-crash)?

Yes, we have tried enabling this as per the documentation.  But the results are the same as before.  There are symbols for standard library calls, but not for our own app code.

For example, this trace is from a build with on-device symbolication enabled (in this case `Booth` is our application):

![Screenshot 2020-01-09 at 08 26 55](https://user-images.githubusercontent.com/1186763/72050821-f3de6b80-32b9-11ea-86b7-f80d351304f1.png)

**Comment by prat14k** [comment]
Hi @akornich, any update?

**Comment by akornich** [comment]
@prat14k , we are currently in the process of planning our workload for the next few months. This issue is getting on the schedule of our backend team. I do not have an exact ETA for the completion but it should be addressed sometime within the next 3 months or so.

**Comment by akornich** [comment]
some (but not all, yet) improvements related to symblication already happened. can someone confirm noticeable improvements?

**Comment by mkilling** [comment]
Unfortunately it does not seem the situation has improved for us.

We still see stack traces for which the symbolication did not translate addresses to method names.

Additionally (and this might be a separate issue) we see occurrences that could not be symbolicated because Rollbar says a .dSYM file has not been uploaded yet, even though it has.

**Comment by akornich** [comment]
@mkilling, thanks for the feedback. May I ask you to either post links to a couple of items having these problems here or to open a support ticket with Rollbar at intercom.com?

**Comment by mkilling** [comment]
I will do that once I’m back in the office. Thanks for your assistance.

**Comment by francois-iveo** [comment]
Have already contacted your support department, but we are experiencing this same issue. Will happily provide information on request.

**Comment by luandrecast-lp** [comment]
any update on this?

**Comment by nihilias** [comment]
Any news regarding this issue? We are still unable to read the crash logs for custom modules.

**Comment by akornich** [comment]
@nihilias , can you, please, provide here a link to the partially or fully non-symbolicated payload?

**Comment by nihilias** [comment]
Sure:
* https://rollbar.com/nanogiants/AktionMensch-LotterieApp/items/1068/
Ping me, if you need more infos or links.

**Comment by akornich** [comment]
@nihilias , thank you! i'll be looking into it today...

**Comment by akornich** [comment]
@nihilias , did you try [enabling on-device symbolication](https://docs.rollbar.com/docs/ios#enabling-on-device-symbolication-from-kscrash)?

**Comment by nihilias** [comment]
It was disabled. We made a branch with enabled settings and will test it as soon as we cann. I'll get back to you with the results 👋🏻

**Comment by nihilias** [comment]
Hey @akornich 
we released an update were on-device symbolication is activated. Still, the crashlogs are not symbolicated. Here is a [link](https://rollbar.com/nanogiants/AktionMensch-LotterieApp/items/2032/).

**Comment by qnoppel** [comment]
Notification for some progress on this: https://github.com/rollbar/rollbar-ios/issues/160#issuecomment-765905503

**Comment by akornich** [comment]
this is a service side issue that was reported a long time ago and was reiterated many times. there is nothing we can/need to change on the SDK side to fix this issue. the backend services are already working on some improvements. closing it here...
